.github/workflows/build-images-docs-builder.yaml

```diff
@@ -0,0 +1,196 @@
+        run: |
+          NEW_IMAGE="quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/docs-builder:${{ needs.build-and-push.outputs.tag }}@${{ needs.build-and-push.outputs.digest }}"
+          # Run in Docker to prevent the script from accessing the environment.
+          docker run --rm -v $PWD:/cilium -w /cilium "${NEW_IMAGE}" \
+              bash -c "git config --global --add safe.directory /cilium && \
+                       /cilium/Documentation/update-docs-builder-image.sh ${NEW_IMAGE}"
+          git commit -sam "ci: update docs-builder"
+
+      - name: Get token
+        id: get_token
+        uses: cilium/actions-app-token@61a6271ce92ba02f49bf81c755685d59fb25a59a # v0.21.1
+        with:
+          APP_PEM: ${{ secrets.AUTO_COMMITTER_PEM }}
+          APP_ID: ${{ secrets.AUTO_COMMITTER_APP_ID }}
+
+      - name: Push changes into PR
+        env:
+          REF: ${{ github.event.pull_request.head.ref }}
+        run: |
+          git diff HEAD^
+          git push https://x-access-token:${{ steps.get_token.outputs.app_token }}@github.com/${{ env.QUAY_ORGANIZATION }}/cilium.git HEAD:"$REF"
+
+  retrieve-digest:
+    name: Retrieve and upload image digest
+    needs: build-and-push
+    if: needs.build-and-push.outputs.digest
+    runs-on: ubuntu-20.04
+    timeout-minutes: 10
+    steps:
+      - name: Checkout default branch (trusted)
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+        with:
+          ref: ${{ github.event.repository.default_branch }}
+          persist-credentials: false
+
+      - name: Set environment variables
+        uses: ./.github/actions/set-env-variables
+
+      - name: Retrieve image digest
+        shell: bash
+        run: |
+          NEW_IMAGE="quay.io/${{ env.QUAY_ORGANIZATION_DEV }}/docs-builder:${{ needs.build-and-push.outputs.tag }}@${{ needs.build-and-push.outputs.digest }}"
+          mkdir -p image-digest/
+          echo "## docs-builder" > image-digest/docs-builder.txt
+          echo "" >> image-digest/docs-builder.txt
+          echo "\`${NEW_IMAGE}\`" >> image-digest/docs-builder.txt
+          echo "" >> image-digest/docs-builder.txt
+
+      - name: Upload artifact digests
+        uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
+        with:
+          name: image-digest docs-builder
+          path: image-digest
+          retention-days: 1
+
+  image-digests:
+    name: Display digests
+    needs: retrieve-digest
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Create image digest directory
+        shell: bash
+        run: |
+          mkdir -p image-digest/
+
+      - name: Download image digest
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          path: image-digest/
+
+      - name: Output image digest
+        shell: bash
+        run: |
+          cd image-digest/
+          find -type f | sort | xargs -d '\n' cat
```

Comment:
Could this step be merged into the previous job?

Reply:
:thinking: Let me see, I didn't pay much attention to that step, but it seems that in the workflow I copied from (bas image generation), we needed a separate job to retrieve images from the different jobs spawned from the matrix; and in our case, we don't need that, given that we only build one image. Is this what you mean?

I'll give it a try

Reply:
OK, seems like [it's still working](https://github.com/cilium/cilium/actions/runs/5799571701/job/15719921971)

<details><summary>Incremental diff</summary>

``` diff
diff --git a/.github/workflows/build-images-docs-builder.yaml b/.github/workflows/build-images-docs-builder.yaml
index b4157a621f0d..fa48b23aa73e 100644
--- a/.github/workflows/build-images-docs-builder.yaml
+++ b/.github/workflows/build-images-docs-builder.yaml
@@ -141,8 +141,8 @@ jobs:
           git diff HEAD^
           git push https://x-access-token:${{ steps.get_token.outputs.app_token }}@github.com/${{ env.QUAY_ORGANIZATION }}/cilium.git HEAD:"$REF"
 
-  retrieve-digest:
-    name: Retrieve and upload image digest
+  image-digest:
+    name: Retrieve and display image digest
     needs: build-and-push
     if: needs.build-and-push.outputs.digest
     runs-on: ubuntu-20.04
@@ -174,21 +174,6 @@ jobs:
           path: image-digest
           retention-days: 1
 
-  image-digests:
-    name: Display digests
-    needs: retrieve-digest
-    runs-on: ubuntu-20.04
-    steps:
-      - name: Create image digest directory
-        shell: bash
-        run: |
-          mkdir -p image-digest/
-
-      - name: Download image digest
-        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
-        with:
-          path: image-digest/
-
       - name: Output image digest
         shell: bash
         run: |
```
</details>

Reply:
Yep that's what I was getting at! The incremental diff looks great!